src/js/pannellum.js

```diff
@@ -392,7 +392,7 @@ function init() {
                 if (xhr.status != 200) {
                     // Display error if image can't be loaded
                     var a = document.createElement('a');
-                    a.href = encodeURI(p);
```

Comment:
This breaks the link and misleads the reason of the error to a wrong URL request.

Reply:
Can you guarantee that this doesn't introduce the potential for DOM XSS attacks? That's why the encoding was added in the first place. My cursory attempt to demonstrate an attack failed, but I'd like to be more confident that this doesn't introduce an attack vector.

Reply:
Good thinking. We need to prevent urls starting with `javascript:` protocol and use `.textContent` instead of `.innerHTML`.

But other types of XSS attacks are a problem only when you're using a template engine on the server. So there is no chance that the browser will get a `<a href="/" onclick=alert('xss')>` when doing `a.href = x`.

Latter we're setting `a.innerHTML` to `a.href`, `a.href` could have HTML tags, so I will change that to `a.textContent = a.href`. Using textContent prevents XSS attacks that could happen with innerHtml ([see rule 6](https://www.owasp.org/index.php/DOM_based_XSS_Prevention_Cheat_Sheet))

Reply:
So I took the liberty to fix other places that could suffer from double encoding URLs and also replaced `encodeURI` with a simple sanitizer to prevent `javascript:` URLs.